plugins/inputs/opensearch_query/opensearch_query.go

```diff
@@ -0,0 +1,309 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package opensearch_query
+
+import (
+	"context"
+	"crypto/tls"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	influxtls "github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/opensearch-project/opensearch-go/v2"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+// OpensearchQuery struct
+type OpensearchQuery struct {
+	URLs                []string        `toml:"urls"`
+	Username            string          `toml:"username"`
+	Password            string          `toml:"password"`
```

Comment:
It would be nice if this new plugin would support secret stores from the beginning. See for example http_response plugin for reference on how to use it.
```suggestion
	Username            config.Secret   `toml:"username"`
	Password            config.Secret   `toml:"password"`
```

Thoughts @srebhan?

Reply:
Done.  That was pretty straightforward to implement.